src/main/java/org/assertj/core/api/BDDAssumptions.java

```diff
@@ -0,0 +1,2392 @@
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
+import java.time.OffsetTime;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicLongArray;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicMarkableReference;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.AtomicReferenceArray;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.concurrent.atomic.AtomicStampedReference;
+import java.util.function.DoublePredicate;
+import java.util.function.IntPredicate;
+import java.util.function.LongPredicate;
+import java.util.function.Predicate;
+import java.util.stream.DoubleStream;
+import java.util.stream.IntStream;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+import org.assertj.core.api.ThrowableAssert.ThrowingCallable;
+import org.assertj.core.util.CheckReturnValue;
+
+/**
+ * Behavior-driven development style entry point for assumption methods for different types, which allow to skip test execution when assumptions are not met.
+ * <p>
+ * The difference with the {@link Assumptions} class is that entry point methods are named <b><code>given</code></b> instead of
+ * <code>assumeThat</code>.
+ * <p>
+ * {@link BDDAssumptions} and {@link BDDAssertions} complement each other to allow a fluent Behavior-driven development.
+ * <p>
+ * Examples:
+ * <pre>{@code
+ * String frodo = "HOBBIT";
+ * List<String> fellowshipOfTheRing = Arrays.asList("Aragorn", "Gandalf", "Frodo", "Legolas"); // and more
+ * }</pre>
+ * <pre><code class='java'>
+ * {@literal @Test}
+ *  public void given_the_assumption_is_not_met_the_test_is_skipped() {
+ *    given(frodo).isEqualTo("ORC");
+ *
+ *    // ... following code is not executed
+ *    then(fellowshipOfTheRing).contains("Sauron");
+ *  }
+ *
+ * {@literal @Test}
+ *  public void given_the_assumption_is_met_the_test_is_executed() {
+ *    given(frodo).isEqualTo("HOBBIT");
+ *
+ *    // ... following code is executed
+ *    then(fellowshipOfTheRing).doesNotContain("Sauron");
+ *  }
+ * </code></pre>
+ *
+ * @since 3.14.0
+ * @author Gonzalo Müller
+ */
+@CheckReturnValue
+public class BDDAssumptions extends Assumptions {
```

Comment:
Add a test to make sure this class will always be in sync with `Assertions` so that if we add another `assertThat` method the test would fail if we forget to add the corresponding `given` method here.

The easiest is to add a  `Assertions_sync_with_BDDAssumptions_Test` test similar to  `Assertions_sync_with_Assumptions_Test` but for `BDDAssumptions`.

Reply:
Done! Thanks!